src/backup_domain.rs

```diff
@@ -22,3 +22,53 @@ use crate::pac::BKP;
 pub struct BackupDomain {
     pub(crate) _regs: BKP,
 }
+
+macro_rules! write_drx {
+    ($self:ident, $drx:ident, $idx:expr, $new:expr) => {
+        unsafe {
+            $self._regs.$drx[$idx].write(|w| w.d().bits($new));
+        }
+    };
+}
+
+macro_rules! read_drx {
+    ($self:ident, $drx:ident, $idx:expr) => {
+        $self._regs.$drx[$idx].read().d().bits();
+    };
+}
+
+impl BackupDomain {
+    /// Read a 16-bit value from one of the DR1 to DR10 registers part of the
+    /// Backup Data Register. The register argument is a zero based index to the
+    /// DRx registers: 0 is DR1, up to 9 for DR10. Providing a number above 9
+    /// will panic.
+    pub fn read_data_register_low(&self, register: usize) -> u16 {
+        read_drx!(self, dr, register)
+    }
+
+    /// Read a 16-bit value from one of the DR11 to DR42 registers part of the
+    /// Backup Data Register. The register argument is a zero based index to the
+    /// DRx registers: 0 is DR11, up to 31 for DR42. Providing a number above 31
+    /// will panic.
+    /// NOTE: not available on medium- and low-density devices!
+    pub fn read_data_register_high(&self, register: usize) -> u16 {
+        read_drx!(self, bkp_dr, register)
+    }
+
+    /// Write a 16-bit value to one of the DR1 to DR10 registers part of the
+    /// Backup Data Register. The register argument is a zero based index to the
+    /// DRx registers: 0 is DR1, up to 9 for DR10. Providing a number above 9
+    /// will panic.
+    pub fn write_data_register_low(&self, register: usize, data: u16) {
+        write_drx!(self, dr, register, data)
+    }
+
+    /// Write a 16-bit value to one of the DR11 to DR42 registers part of the
+    /// Backup Data Register. The register argument is a zero based index to the
+    /// DRx registers: 0 is DR11, up to 31 for DR42. Providing a number above 31
+    /// will panic.
+    /// NOTE: not available on medium- and low-density devices!
+    pub fn write_data_register_high(&self, register: usize, data: u16) {
+        write_drx!(self, dr, register, data)
```

Comment:
write_drx!(self, bkp_dr, register, data)

Reply:
Sorry, I'm happy that you're more alert than I am!